anesthetic/plotting/_matplotlib/misc.py

```diff
@@ -0,0 +1,14 @@
+import pandas.plotting._matplotlib.misc as misc
+from anesthetic.plotting._matplotlib.core import _compress_weights
+
+
+def scatter_matrix(frame, *args, **kwargs):
+    # noqa: disable=D103
+    frame = _compress_weights(kwargs, frame)
```

Comment:
Compute a weighted scatter matrix by compressing to the channel capacity (could technically do slightly better than this by incorporating weights into the KDE/hist diagonal, but this would involve re-writing the whole function, when we have more functionality in plot_2d

Reply:
Where does this get used? 

> could technically do slightly better than this by incorporating weights into the KDE/hist diagonal

Not sure what you mean here.

Reply:
`scatter_matrix` is the pandas equivalent of our `plot_2d` (but with less functionality). it is used in `tests.test_weighted_pandas.test_scatter_matrix`. Implementation wise I could be slightly smarter by incorporating weights for the diagonal plots of the kde and hist, rather than just compressing to equally weighted samples as we do at the moment.

```python
>>> from pandas.plotting import scatter_matrix
>>>
>>> scatter_matrix(wdf)                                                                      
```

Reply:
Hmm, difficult one. I'm fine either way. Maybe not so bad to have the 1D hist be the projection from the 2D scatter realisation.

On another note, how does `scatter_matrix` handle the split into 1D diagonal plots and 2D off-diagonal plots? Is there something better than our approach with `twin` axes?

Reply:
To recount our analogue discussion just now -- I think that (a) scatter_plot is not a particularly good way to do it (b) it would be good to revisit whether the `twin` axes is the best way to achieve what we want given that (c) the axes twinning is primarily designed to ensure that yticks appear on the top left corner of a plot_2d.

Reply:
To recount our analogue discussion just now -- I think that (a) scatter_plot is not a particularly good way to do it (b) it would be good to revisit whether the `twin` axes is the best way to achieve what we want given that (c) the axes twinning is primarily designed to ensure that yticks appear on the top left corner of a plot_2d.